plugins/in_tail/tail.c

```diff
@@ -144,6 +144,11 @@ static int in_tail_collect_static(struct flb_input_instance *i_ins,
                           file->name);
                 flb_tail_file_remove(file);
             }
+            if (file->config->exit_on_eof) {
```

Comment:
would you please move this routine before the file gets promoted  (few lines back)